mods/farming/api.lua

```diff
@@ -319,11 +319,9 @@ farming.register_plant = function(name, def)
 		nodegroups[pname] = i
 
 		local next_plant = nil
-		local on_timer = nil
 
 		if i < def.steps then
 			next_plant = mname .. ":" .. pname .. "_" .. (i + 1)
-			on_timer = farming.grow_plant
```

Comment:
As Explanation:
This assigns `farming.grow_plant` to a local variable, that is not used afterwards, because in line 343 it is again assigned to the callback, skipping the local `on_timer` variable.

But while i can guarantee that this is matching the behavior of current master; it would indeed be nice to hear from @sofar if this is also intended behavior, or if line 343 should use `on_timer` instead.